hnn_core/network.py

```diff
@@ -299,6 +299,40 @@ def pick_connection(net, src_gids=None, target_gids=None,
     return sorted(conn_set)
 
 
+def _get_cell_index_by_synapse_type(net):
```

Comment:
Is there a reason to put this outside of the `Network` class as a standalone function if it requires an existing `Network`?

Reply:
my general rule is to keep methods of a class minimal, so the structure of the class/object is clear. Just because it takes the object as input doesn't mean it should be a method. The method should have an explicit purpose. For example:

```py
ica = ICA()
ica.fit(X)
y = ica.predict(X)
```

once something becomes a method, it's very hard to disentangle it from the class